backend/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSender.java

```diff
@@ -0,0 +1,132 @@
+package com.redhat.cloud.notifications.processors.email;
+
+import com.redhat.cloud.notifications.db.EndpointResources;
+import com.redhat.cloud.notifications.ingress.Action;
+import com.redhat.cloud.notifications.models.EmailSubscriptionProperties;
+import com.redhat.cloud.notifications.models.Notification;
+import com.redhat.cloud.notifications.models.NotificationHistory;
+import com.redhat.cloud.notifications.processors.email.bop.Email;
+import com.redhat.cloud.notifications.processors.email.bop.Emails;
+import com.redhat.cloud.notifications.processors.webclient.SslVerificationDisabled;
+import com.redhat.cloud.notifications.processors.webhooks.WebhookTypeProcessor;
+import com.redhat.cloud.notifications.recipients.User;
+import io.quarkus.qute.TemplateInstance;
+import io.smallrye.mutiny.Uni;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mutiny.core.buffer.Buffer;
+import io.vertx.mutiny.ext.web.client.HttpRequest;
+import io.vertx.mutiny.ext.web.client.WebClient;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+@ApplicationScoped
+public class EmailSender {
+
+    private final Logger logger = Logger.getLogger(this.getClass().getName());
+
+    static final String BOP_APITOKEN_HEADER = "x-rh-apitoken";
+    static final String BOP_CLIENT_ID_HEADER = "x-rh-clientid";
+    static final String BOP_ENV_HEADER = "x-rh-insights-env";
+    static final String BODY_TYPE_HTML = "html";
+
+    @Inject
+    @SslVerificationDisabled
+    WebClient unsecuredWebClient;
+
+    @ConfigProperty(name = "processor.email.bop_url")
+    String bopUrl;
+
+    @ConfigProperty(name = "processor.email.bop_apitoken")
+    String bopApiToken;
+
+    @ConfigProperty(name = "processor.email.bop_client_id")
+    String bopClientId;
+
+    @ConfigProperty(name = "processor.email.bop_env")
+    String bopEnv;
+
+    @ConfigProperty(name = "processor.email.no_reply")
+    String noReplyAddress;
+
+    @Inject
+    WebhookTypeProcessor webhookSender;
+
+    @Inject
+    EndpointResources endpointResources;
+
+    public Uni<NotificationHistory> sendEmail(User user, Action action, TemplateInstance subject, TemplateInstance body) {
```

Comment:
If we need to send the same email to multiple recipients, will this method be called once for each recipient?

Reply:
Found the (obvious) answer after reading the Jira ticket: emails can be personalized so each recipient is processed separately.

Did we consider the potentially huge increase of emails we will send through the BOP? It would be interesting to know the average of bcc recipients per email currently sent on prod. We have to be extra careful when this goes to prod. The injected `WebClient` instance has a fixed capacity that may be exceeded which could lead to slow/delayed BOP queries.

Reply:
The following instance is shared with unsecured webhook calls:
```
@Inject
    @SslVerificationDisabled
    WebClient unsecuredWebClient;
```
We should separate webhooks from emails in this PR and produce an additional `WebClient` instance that would be dedicated to emails only. That way, if we experience poor performances because of personalized emails, webhooks won't be impacted.

Reply:
It would be great if we could monitor with metrics the average execution time between the moment we want to send an email and the moment when the BOP request is actually completed.

Reply:
> Did we consider the potentially huge increase of emails we will send through the BOP? It would be interesting to know the average of bcc recipients per email currently sent on prod. We have to be extra careful when this goes to prod. The injected `WebClient` instance has a fixed capacity that may be exceeded, which could lead to slow/delayed BOP queries.

One *optimization* we could eventually apply here is to batch emails. BOP allows to make one call with multiple emails.

Reply:
That would be great. Let's do that in follow-up PR.

Reply:
https://issues.redhat.com/browse/NOTIF-271